ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -73,9 +73,18 @@ void show_error(string errortext, const bool fatal)
 }
 
 namespace enigma {
-  extern HINSTANCE hInstance;
-  extern HWND hWnd;
-  HWND infore;
+
+extern HINSTANCE hInstance;
+extern HWND hWnd;
+HWND infore;
+
+}
+
+static inline string remove_slash(string dir) {
```

Comment:
why does add_slash take a const reference but remove doesn't. Do you even know what const ref is?

Reply:
const's can't be changed. I made the one that needed to be modified able to do so. I thought you knew this, but apparently you need to write me a paragraph essay on the difference because you were the one who doesn't know.

Reply:
neither need to be modified as you are returning a brand new string in both rather than modifying the existing one.

Reply:
Indeed, you can accept `const&` here and return `dir` or `dir.substr(...)`.